modules/internal/ChapelDistribution.chpl

```diff
@@ -394,5 +582,63 @@ module ChapelDistribution {
     proc dsiSupportsBulkTransferInterface() param return false;
     proc doiCanBulkTransferStride() param return false;
   }
-  
+
+  class BaseSparseArr: BaseArr {
+    type eltType;
+    param rank : int;
+    type idxType;
+
+    var dom; /* : DefaultSparseDom(?); */
+    /*var dataDom = {1..0};*/
+    var data: [dom.nnzDom] eltType;
```

Comment:
Also, I thought about making `dom` of `BaseSparseArr` a `BaseSparseDom`. However, `BaseSparseDom` has a generic `parentDom` which we don't have access to at that point.

This would have looked like

```
class BaseSparseArr: BaseArr {
   // ....
   var dom: BaseSparseDom(rank, idxType, parentDom); //but what is parentDom?
   // ....
}
```